### PR TITLE
create_perf_json: Drop UMaskExt depending on FCMask and PortMask

### PR DESCRIPTION
**Pull request by edwarddavidbaker**
This commit **unsets** `UMaskExt` for events which also populate `PortMask` or `FCMask`.

### CODE REVIEW

**Comment by edwarddavidbaker** [comment]
This commit is intended to update uncore IIO events. The following diff compares `create_perf_json.py` output from main and this pull request.

Output files which are then modified:
```
emeraldrapids/uncore-io.json
grandridge/uncore-io.json 
graniterapids/uncore-io.json 
sapphirerapids/uncore-io.json 
sierraforest/uncore-io.json
```

```diff
$ diff -rup -U8 main/ iio-fix/ | head -n100
diff -rup -U8 main/emeraldrapids/uncore-io.json iio-fix/emeraldrapids/uncore-io.json
--- main/emeraldrapids/uncore-io.json   2024-12-04 14:07:18.610906985 -0700
+++ iio-fix/emeraldrapids/uncore-io.json        2024-12-04 14:21:08.666139989 -0700
@@ -116,108 +116,108 @@
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART0",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0001",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 0 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 0",
-        "UMask": "0x7001004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
     {
         "BriefDescription": "PCIe Completion Buffer Inserts of completions with data: Part 1",
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART1",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0002",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 1 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 1",
-        "UMask": "0x7002004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
     {
         "BriefDescription": "PCIe Completion Buffer Inserts of completions with data: Part 2",
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART2",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0004",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 2 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 2",
-        "UMask": "0x7004004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
     {
         "BriefDescription": "PCIe Completion Buffer Inserts of completions with data: Part 3",
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART3",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0008",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 2 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 3",
-        "UMask": "0x7008004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
     {
         "BriefDescription": "PCIe Completion Buffer Inserts of completions with data: Part 4",
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART4",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0010",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 0 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 4",
-        "UMask": "0x7010004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
     {
         "BriefDescription": "PCIe Completion Buffer Inserts of completions with data: Part 5",
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART5",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0020",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 1 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 5",
-        "UMask": "0x7020004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
     {
         "BriefDescription": "PCIe Completion Buffer Inserts of completions with data: Part 6",
         "Counter": "0,1,2,3",
         "EventCode": "0xc2",
         "EventName": "UNC_IIO_COMP_BUF_INSERTS.CMPD.PART6",
         "Experimental": "1",
         "FCMask": "0x07",
         "PerPkg": "1",
         "PortMask": "0x0040",
         "PublicDescription": "PCIe Completion Buffer Inserts of completions with data : Part 2 : x16 card plugged in to Lane 0/1/2/3, Or x8 card plugged in to Lane 0/1, Or x4 card is plugged in to slot 6",
-        "UMask": "0x7040004",
+        "UMask": "0x4",
         "Unit": "IIO"
     },
<snip>
```

File: scripts/create_perf_json.py [hunk 1]
```diff
@@ -327,6 +327,13 @@ def get(key: str) -> str:
         if self.umask:
             self.umask = self.umask.split(",")[0]
             umask_ext = get('UMaskExt')
+
```
Comment (captain5050) [inline scripts/create_perf_json.py:330]: This change will apply the fix to more than just IIO. I had the following:
```
            if umask_ext and self.unit != 'IIO':
                self.umask = umask_ext + self.umask[2:]
```

Comment (edwarddavidbaker) [reply scripts/create_perf_json.py:330]: Thanks Ian, that certainly works as well. I'm getting the same output with both approaches, and yours is certainly simpler.

`PortMask` and `FCMask` might be removed on subsequent platforms. We could revisit this logic at that point.

Comment (captain5050) [reply scripts/create_perf_json.py:330]: Thanks for checking! I'm happy to drop my fix in favor of this.

File: scripts/create_perf_json.py [hunk 2]
```diff
@@ -731,7 +738,7 @@ def save_form(self, name: str, group: str, form: str, desc: str, locate: str,
                      'num_packages', 'num_cores', 'SYSTEM_TSC_FREQ',
                      'filter_tid', 'TSC', 'cha', 'config1',
                      'source_count', 'slots', 'thresh', 'has_pmem',
-                     'num_dies', 'num_cpus_online', 'PEBS', 'R',
+                     'num_dies', 'num_cpus_online', 'PEBS', 'pcu_0', 'R',
```
Comment (captain5050) [inline scripts/create_perf_json.py:741]: This change looks unrelated and shouldn't be part of the commit.

**Review by captain5050 — APPROVED** [review]

File: scripts/create_perf_json.py (hunk 1 above)
Comment (edwarddavidbaker) [reply scripts/create_perf_json.py:330]: Thanks Ian. Hopefully these changes don't introduce too many conflicts in your local `create_perf_json.py` version!
